2386/2/PopularWords.rb

```diff
@@ -0,0 +1,42 @@
+# :reek:FeatureEnvy
+# :reek:NestedIterators
+# :reek:UtilityFunction
+# :reek:TooManyStatements
+# :reek:IrresponsibleModule
+# PopularWords
+class PopularWords
+  def hash_all_words(name)
+    search_battles(name).each_with_object(Hash.new(0)) do |word, word_count|
+      File.open("Rapbattle/#{word}", 'r').each_line do |line|
+        next if line["/Раунд \d/i"]
+        line.split(' ').select { |each_word| wort_test(each_word) }
+            .select do |term|
+          word_count[term] += 1
+        end
+      end
+    end
+  end
+
+  def prepare_hash(raper_name)
+    hash = hash_all_words(raper_name).keys.join(' ')
+                                     .downcase
+                                     .gsub(/[.,!?:;«»<>&'()] /, ' ')
```

Comment:
you replace these symbols with whitespace and then split by whitespace...
Don't you want to split by these symbols instead? Just add whitespace to the list.